content/vendor/guides/airgapped-existing-cluster.md

```diff
@@ -0,0 +1,523 @@
+* TCP_NODELAY set
+* Immediate connect fail for 2607:f8b0:4001:c05::64: Network is unreachable
+  0     0    0     0    0     0      0      0 --:--:--  0:00:03 --:--:--     0
+```
+
+Once you're satisfied the instance has no outbound connectivity, you can Ctrl+C this command and we'll proceed to setting up our airgapped cluster.
+
+
+#### Airgapped cluster with registry
+
+Next, we'll create a GCP vm with online internet access to be our airgapped cluster. As before, we'll use a an internet connection to install k8s and get a registry up and running before removing access.
+
+```shell script
+gcloud compute instances create ${AIRGAP_CLUSTER} \
+  --boot-disk-size=200GB \
+  --image-project ubuntu-os-cloud \
+  --image-family ubuntu-1804-lts \
+  --machine-type n1-standard-4
+```
+
+ Before installing Docker and Kubernetes, let's get the private IP and set it as an insecure docker registry
+
+```shell script
+export CLUSTER_PRIVATE_IP=$( \
+  gcloud compute instances describe ${AIRGAP_CLUSTER} \
+  --format='get(networkInterfaces[0].networkIP)')
+# verify
+echo ${CLUSTER_PRIVATE_IP}
+```
+
+```shell script
+ gcloud compute ssh ${AIRGAP_CLUSTER} -- "sudo mkdir -p /etc/docker"
+ gcloud compute ssh ${AIRGAP_CLUSTER} -- \
+  "echo \"{\\\"insecure-registries\\\":[\\\"${CLUSTER_PRIVATE_IP}:32000\\\"]}\" | sudo tee /etc/docker/daemon.json"
+```
+
+
+Now, let's ssh into the instance and bootstrap a minimal single-primary kubernetes cluster (details here:  https://kurl.sh/1010f0a  )
+
+```shell script
+gcloud compute ssh ${AIRGAP_CLUSTER} -- 'curl  https://k8s.kurl.sh/1010f0a  | sudo bash'
+```
+
+When that completes, we can deploy a minimal registry and verify it's running
+
+```shell script
+gcloud compute ssh ${AIRGAP_CLUSTER} -- \
+  'kubectl --kubeconfig ./admin.conf apply -f https://gist.githubusercontent.com/dexhorthy/7a3e6eb119d2d90ff7033a78151c3be2/raw/6c67f95367988d1a016635e3da689e2d998d458c/plain-registry.yaml'
+```
+
+This gist configures a basic auth htpasswd that configures a username/password for `kots/kots`, which we'll use later
+
+```shell script
+gcloud compute ssh ${AIRGAP_CLUSTER} -- \
+  'kubectl --kubeconfig ./admin.conf get pod,svc -n registry'
+```
+
+Now that the registry is up, let's verify that we can docker push/pull to it. We'll use the public IP attached to the instance.
+
+```shell script
+export CLUSTER_PUBLIC_IP=$( \
+  gcloud compute instances describe ${AIRGAP_CLUSTER} \
+  --format='get(networkInterfaces[0].accessConfigs[0].natIP)')
+docker login --username kots --password kots ${CLUSTER_PUBLIC_IP}:32000
+docker pull busybox
+docker tag busybox ${CLUSTER_PUBLIC_IP}:32000/busybox
+docker push ${CLUSTER_PUBLIC_IP}:32000/busybox
+```
+
+you may need to also add an `insecure-registy` entry to your workstation to allow pushing/pulling via http instead of https. If you're testing using docker-for-mac, you can add this via the setttings:
```

Comment:
```suggestion
You may need to also add an `insecure-registy` entry to your workstation to allow pushing/pulling via http instead of https. If you're testing using docker-for-mac, you can add this via the setttings:
```